python/test/gluon/test_frontend.py

```diff
@@ -880,6 +880,27 @@ def test_split_join():
     ttgl.static_assert(d.type.layout == layout)
 
 
+@filecheck_test
+@gluon.jit
+def test_split_join_subtile():
+    # CHECK-DAG: [[LAYOUT:#.*]] = #ttg.blocked<{sizePerThread = [1, 128], threadsPerWarp = [32, 1], warpsPerCTA = [4, 1], order = [0, 1]}>
+    # CHECK-DAG: [[JOIN_LAYOUT:#.*]] = #ttg.blocked<{sizePerThread = [1, 64, 2], threadsPerWarp = [32, 1, 1], warpsPerCTA = [4, 1, 1], order = [0, 1, 2]}>
+    # CHECK-DAG: [[LINEAR:#.*]] = #ttg.linear
+    layout: ttgl.constexpr = ttgl.BlockedLayout([1, 128], [32, 1], [4, 1], [0, 1])
+    x = ttgl.full([128, 128], 1, ttgl.int32, layout=layout)
+
+    a, b = x.reshape([128, 2, 64]).permute([0, 2, 1]).split()
+    join_layout: ttgl.constexpr = ttgl.BlockedLayout([1, 64, 2], [32, 1, 1], [4, 1, 1], [0, 1, 2])
+    # CHECK: tt.join {{.*}} : tensor<128x64xi32, #{{.*}}> -> tensor<128x64x2xi32, [[JOIN_LAYOUT]]>
+    y = ttgl.join(a, b, layout=join_layout).permute([0, 2, 1]).reshape([128, 128])
+
+    # CHECK: arith.addi {{.*}} : tensor<128x128xi32, [[LAYOUT]]>
+    _ = x + y
+
+    # CHECK: tt.reshape {{.*}} : tensor<128x2x64xi32, #{{.*}}> -> tensor<128x128xi32, [[LINEAR]]>
```

Comment:
I see, so the layouts are equivalent you just need it to be a blocked layout not a linear layout. Seems like it would be better to fix the layout inference to preserve block layouts. In the mean time you can just use `convert_layout` to get a blocked layout.